crates/holochain/src/core/ribosome/host_fn/must_get_entry.rs

```diff
@@ -0,0 +1,269 @@
+use holochain_wasmer_host::prelude::*;
+use crate::core::ribosome::CallContext;
+use crate::core::ribosome::HostContext;
+use crate::core::ribosome::RibosomeT;
+use holochain_cascade::Cascade;
+use holochain_types::prelude::*;
+use holochain_wasmer_host::prelude::WasmError;
+use std::sync::Arc;
+use holochain_p2p::actor::GetOptions as NetworkGetOptions;
+use holochain_p2p::event::GetRequest;
+
+#[allow(clippy::extra_unused_lifetimes)]
+pub fn must_get_entry<'a>(
+    _ribosome: Arc<impl RibosomeT>,
+    call_context: Arc<CallContext>,
+    input: MustGetEntryInput,
+) -> Result<EntryHashed, WasmError> {
+    let entry_hash = input.into_inner();
+    let network = call_context.host_context.network().clone();
+
+    // timeouts must be handled by the network
+    tokio_helper::block_forever_on(async move {
+        let workspace = call_context.host_context.workspace();
+        let mut cascade = Cascade::from_workspace_network(workspace, network);
+        match cascade
+            .retrieve_entry(entry_hash.clone(),
```

Comment:
Just pointing out that `retrieve_*` was never designed for use in host_fn. Art didn't want it available to the wasm. Note that this actually gets the entry **before** it's validated. So it's not just invalid / valid ops that are returned but also "not yet validated". I think you should check with Art if this is ok.

Reply:
I asked Art about this and it doesn't seem to be an issue.

Reply:
Makes sense since these are just about getting content regardless of validation status

Reply:
@freesig yeah we had a kind of long conversation about this and decided always to return it even if not validated